src/components/toggle.jsx

```diff
@@ -7,33 +7,36 @@ class ToggleContainerViews extends React.Component {
   constructor(props) {
     super();
 
-    const view = props.defaultView;
+    const view = props.defaults.view;
 
     this.state = {
       view,
     };
   }
 
   handleClick(view) {
-    this.setState({
+    this.props.store.dispatch({
+      type: 'toggle',
+      async: false,
       view,
     });
   }
 
   render() {
     return (
       <div>
-        <button onClick={() => { this.handleClick('NearYou'); }}>NearYou</button>
-        <button onClick={() => { this.handleClick('MyAds'); }}>MyAds</button>
-        <button onClick={() => { this.handleClick('Search'); }}>Search</button>
-        <HandleView view={this.state.view} />
+        <button {...this.props} onClick={() => { this.handleClick('NearYou'); }}>NearYou</button>
+        <button {...this.props} onClick={() => { this.handleClick('MyAds'); }}>MyAds</button>
```

Comment:
The buttons aren't react components and so don't need to be passed {...this.props}